testing/trino-server-dev/src/main/java/io/trino/server/HybridDevelopmentPluginsProvider.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.server;
+
+import com.google.inject.Inject;
+import io.trino.server.PluginManager.PluginsProvider;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Supports loading internal and external plugins simultaneously during development.
+ * The `plugin.dir` value in `config.properties` is relative to `trino-server-dev` module.
```

Comment:
This should be added to the docs maybe

Reply:
and we should clarify if this is an additional dir or a replacement and how to set it up in detail

Reply:
Do you think we should add some documentation about this to the main `README` where `DevelopmentServer` is discussed? That's the only place we have any docs about `DevelopmentServer` currently, AFAICT, though I'm not sure if this is worth discussing in the main `README`.

https://github.com/trinodb/trino/blob/bbc15d3fde9abef35ed3058e066c4efbee89789e/README.md?plain=1#L104-L118

Reply:
Yeah.. I think it belongs there .. we dont have a better place at the moment .. in the long run it should be part of https://trino.io/docs/current/develop.html .. but we can tackle that later separately.

Reply:
Sounds good, moved!